examples/cfn-custom-resource/index.ts

```diff
@@ -0,0 +1,98 @@
+// Copyright 2016-2024, Pulumi Corporation.
+
+import * as pulumi from '@pulumi/pulumi';
+import * as aws from "@pulumi/aws-native";
+import * as awsClassic from "@pulumi/aws";
+
+const amiRegion = new pulumi.Config().require("amiRegion");
+
+// Create an IAM role for the Lambda function
+const lambdaRole = new awsClassic.iam.Role("lambdaRole", {
+    assumeRolePolicy: awsClassic.iam.assumeRolePolicyForPrincipal({ Service: "lambda.amazonaws.com" }),
+});
+
+const policy = new awsClassic.iam.Policy("lambdaPolicy", {
+    policy: {
+        Version: "2012-10-17",
+        Statement: [{
+            Action: "ec2:DescribeImages",
+            Effect: "Allow",
+            Resource: "*",
+        }],
+    },
+});
+
+const rpa1 = new awsClassic.iam.RolePolicyAttachment("lambdaRolePolicyAttachment1", {
+    role: lambdaRole.name,
+    policyArn: policy.arn,
+});
+
+const rpa2 = new awsClassic.iam.RolePolicyAttachment("lambdaRolePolicyAttachment2", {
+    role: lambdaRole.name,
+    policyArn: awsClassic.iam.ManagedPolicies.AWSLambdaBasicExecutionRole,
+});
+
+const bucket = new awsClassic.s3.Bucket('custom-resource-emulator', {
+    forceDestroy: true,
+});
+
+const handlerCode = new awsClassic.s3.BucketObjectv2("handler-code", {
+    bucket: bucket.bucket,
+    key: "handlerCode",
+    source: new pulumi.asset.AssetArchive({
+        "index.js": new pulumi.asset.FileAsset("ami-lookup.js"),
+    })
+})
+
+// Create the Lambda function for the custom resource
+const lambdaFunction = new awsClassic.lambda.Function("ami-lookup-custom-resource", {
+    runtime: awsClassic.types.enums.lambda.Runtime.NodeJS16dX,
+    s3Bucket: bucket.bucket,
+    s3Key: handlerCode.key,
+    handler: "index.handler",
+    role: lambdaRole.arn,
+    memorySize: 128,
+    timeout: 30,
+}, { dependsOn: [rpa1, rpa2] });
+
+const cfnCustomResource = new aws.cloudformation.CustomResourceEmulator('emulator', {
+    bucketName: bucket.id,
+    bucketKeyPrefix: 'custom-resource-emulator',
+    customResourceProperties: {
+        Region: amiRegion,
+        Architecture: 'HVM64',
+    },
+    serviceToken: lambdaFunction.arn,
```

Comment:
Ahhh so serviceToken is not called lambdaFunctionARN because some custom resources are backed by something else?

Reply:
Yeah, there's also SNS backed Custom Resources: https://github.com/pulumi/pulumi-aws-native/issues/1812

They're not widely used, so we decided to skip support for them for now. It's not in scope of the CDK work

Reply:
The naming makes a bit more sense now.